### PR TITLE
Restore derived metrics using new metrics

### PR DESCRIPTION
**Pull request by cmcfarlen**
traffic_manager used to manage some derived metrics (metrics calculated from other metrics).  This code was removed with traffic_manager.  This PR restores that functionality and the removed metrics.

### CODE REVIEW

File: include/tsutil/Metrics.h [hunk 1]
```diff
@@ -506,6 +507,18 @@ class Metrics
 
   }; // class Counter
 
+  class Derived
+  {
```
Comment (bneradt) [inline include/tsutil/Metrics.h:511]: Let's add doxygen comments explaining the responsibilities and purpose of this class.

File: include/tsutil/Metrics.h [hunk 2]
```diff
@@ -506,6 +507,18 @@ class Metrics
...
+  public:
+    struct DerivedMetricSpec {
+      std::string_view                                                                              derived_name;
+      std::initializer_list<std::variant<Metrics::AtomicType *, Metrics::IdType, std::string_view>> derived_from;
```
Comment (bneradt) [inline include/tsutil/Metrics.h:515]: A using alias for the variant could be helpful here.

File: include/tsutil/Metrics.h [hunk 3]
```diff
@@ -506,6 +507,18 @@ class Metrics
...
+    };
+
+    static void derive(const std::initializer_list<DerivedMetricSpec> &metrics);
+    static void update_derived();
```
Comment (bneradt) [inline include/tsutil/Metrics.h:519]: doxygen

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good.

**Review by bneradt — APPROVED** [review]

**Comment by cmcfarlen** [comment]
https://github.com/apache/trafficserver/blob/9.2.x/mgmt/DerivativeMetrics.cc

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**